lte/gateway/c/oai/lib/s6a_proxy/S6aClient.cpp

```diff
@@ -88,17 +90,63 @@ static bool read_mme_cloud_subscriberdb_enabled(void) {
   return mconfig.cloud_subscriberdb_enabled();
 }
 
-S6aClient& S6aClient::get_instance() {
-  static S6aClient client_instance;
-  return client_instance;
+S6aClient& S6aClient::get_s6a_proxy_instance(bool enable_s6a_proxy_channel) {
+  static S6aClient s6a_proxy_instance(enable_s6a_proxy_channel);
+  return s6a_proxy_instance;
 }
 
-S6aClient::S6aClient() {
-  // Create channel based on relay_enabled and cloud_subscriberdb_enabled
-  // flags. If relay_enabled is true, then create a channel towards the FeG.
+S6aClient& S6aClient::get_subdb_instance(bool enable_s6a_proxy_channel) {
+  static S6aClient subdb_instance(enable_s6a_proxy_channel);
+  return subdb_instance;
```

Comment:
Why do we need `enable_s6a_proxy_channel` as a boolean argument here. Isn't it always false for subdb_instance?
```suggestion
S6aClient& S6aClient::get_subdb_instance() {
  static S6aClient subdb_instance(false);
  return subdb_instance;
```